tokio/src/task/blocking.rs

```diff
@@ -1,8 +1,10 @@
 use crate::task::JoinHandle;
 
 cfg_rt_multi_thread! {
-    /// Runs the provided blocking function on the current thread without
-    /// blocking the executor.
+    /// Runs the provided blocking function on the current thread (synchronous
+    /// context) without blocking the executor. This function could be a
+    /// replacement for [`thread::spawn`] in cases where a `join` is needed
+    /// but will block the executor whereas this does not.
```

Comment:
If we are to talk about `thread::spawn`, I don't really want it to be in this manner. There are many ways that people try to circumvent it, and `thread::spawn` is only one of them. Perhaps instead a section that says something like "in an attempt to block within the async context, people sometimes do this ..., but you can do it like this instead ..."

Reply:
```suggestion
    /// Runs the provided blocking function on the current thread (synchronous
    /// context) without blocking the executor. In an attempt to block within the
    /// async context, people sometimes do [`thread::spawn`] with `join` but you
    /// can do it with `block_in_place` where it does not block the executor.
```

Will this be better?

Reply:
A bit, though I think the sentence is a bit awkward. I think it would help if it is in its own paragraph.

Reply:
```suggestion
    /// Runs the provided blocking function on the current thread (synchronous
    /// context) without blocking the executor.
    ///
    /// In an attempt to block within the async context, people sometimes do
    /// [`thread::spawn`] with `join` but it can be done with [`block_in_place`]
    /// which does not block the executor.
```